cachecontrol/controller.py

```diff
@@ -58,8 +59,28 @@ def _urlnorm(cls, uri):
         return defrag_uri
 
     @classmethod
-    def cache_url(cls, uri):
-        return cls._urlnorm(uri)
+    def _authorization_digest(cls, authorization_header):
+        """Creates a hash digest of a request's authorization header.
+        """
+        if not authorization_header:
+            return ''
+        return hashlib.sha224(authorization_header.encode('utf-8')).hexdigest()
+
+    @classmethod
+    def cache_key(cls, request, response=None):
+        """Create a cache_key from a url and request headers.
+
+        Requests are generally stored by their URL.
+        For requests that have an Authorization header, we add a hash of that header to the cache key.
+        The hash is only used to make sure we allow caching for each user - it's not a security feature.
+        """
+        normalized_url = cls._urlnorm(request.url)
+        authorization_hash = cls._authorization_digest(request.headers.get('Authorization'))
+        vary_values = response.headers.get('vary', '') if response and response.headers else ''
```

Comment:
See above, I think this can all go away - also it looks a bit weird that we first create the hash, then check the response headers and then potentially remove the hash again, but this will go away anyway